packages/react-server/src/ReactFizzServer.js

```diff
@@ -2456,3 +2484,18 @@ export function abort(request: Request, reason: mixed): void {
     fatalError(request, error);
   }
 }
+
+// If new resources are identified out of band with a task we can still potentially
+// emit them early by pinging them.
+export function pingRequest(): void {
+  const request = resolveCurrentRequest();
+  if (request) {
```

Comment:
If `request.destination === null` we don't have to  schedule work because there's nothing to flush too. That's an assumption we can make because we know we're not going to work on anything.

In fact, really we should probably just be scheduling `flushCompletedQueues` since we're not doing any work but that's maybe over optimizing.

Reply:
Refactored this a bit. Ended up just flushing queues/chunks because the flush can run outside of the dispatcher and context scoping that happens during performWork so it must not rely on it.